x-pack/plugins/cloud_security_posture/server/routes/benchmark_rules/bulk_action/v1.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+import { Logger } from '@kbn/core/server';
+import { CspBenchmarkRules } from '../../../../common/types/rules/v3';
+import { buildRuleKey, getCspSettingsSafe, setRulesStates, updateRulesStates } from './utils';
+
+const muteStatesMap = {
+  mute: true,
+  unmute: false,
+};
+
+export const bulkActionBenchmarkRulesHandler = async (
+  encryptedSoClient: SavedObjectsClientContract,
+  rulesToUpdate: CspBenchmarkRules,
+  action: 'mute' | 'unmute',
+  logger: Logger
+) => {
+  const cspSettings = await getCspSettingsSafe(encryptedSoClient, logger);
```

Comment:
nit: consider using partial update and `upsert`

Reply:
Done, and it saves a lot of code since now we don't need to take care for checking if the object exists:). 
Nice catch:)